### PR TITLE
Fix resting with full hp and a status effect message

### PR DESCRIPTION
**Pull request by Volco**
Resting with full hp + a status now displays "'s HP is full!" and not "But it failed!"

### CODE REVIEW

**Comment by Zarel** [comment]
Messages are implemented client-side (so they can be translated).

File: data/moves.js [hunk 1]
```diff
@@ -13516,7 +13516,11 @@ let BattleMovedex = {
 		flags: {snatch: 1, heal: 1},
 		onTryMove(pokemon) {
 			if (pokemon.hp < pokemon.maxhp && pokemon.status !== 'slp' && !pokemon.hasAbility('comatose')) return;
-			this.add('-fail', pokemon);
+			if (pokemon.hp === pokemon.maxhp && pokemon.status) {
+				this.add('-message', pokemon.name + "'s HP is full!");
```
Comment (Zarel) [inline data/moves.js]: In other words, this should be:

```suggestion
				this.add('-fail', pokemon, 'move: Rest');
```

**Comment by Zarel** [comment]
All this is assuming this is the correct error message. @Marty-D? @DaWoblefet?

File: data/moves.js (hunk 1 above)
Comment (Volco) [reply data/moves.js]: Oh there was a fail message for rest added to the client? Good to know

**Comment by Volco** [comment]
I got it from the known bug mechanics project so it should be

**Comment by scheibo** [comment]
https://github.com/Zarel/Pokemon-Showdown/projects/3#card-24620941

File: data/moves.js (hunk 1 above)
Comment (Zarel) [reply data/moves.js]: No, but you (or someone else) should add it to the client.

**Comment by Zarel** [comment]
Sounds good. I wonder if there's some way to remind users to link Known Mechanics Bugs in relevant PRs.

File: data/moves.js (hunk 1 above)
Comment (Volco) [reply data/moves.js]: ah

Comment (Volco) [reply data/moves.js]: i'll look at the client and see if i can figure out how to do that change. shouldn't be too difficult i'd imagine

**Comment by scheibo** [comment]
You could edit it into the main post, but you could also just do a search in the mechanics bug project if you see a PR around mechanics and are unsure of the context...

**Comment by Volco** [comment]
ill edit the prs i have with a link to the relevant mechanic bugs

File: data/moves.js (hunk 1 above)
Comment (Zarel) [reply data/moves.js]: You just need to add

```
fail: "[POKEMON]'s HP is full!",
```

to `data/text.js`.

Comment (Volco) [reply data/moves.js]: oh, okay thanks. that makes it easier since i'd have to have found the correct file

**Comment by Marty-D** [comment]
The HP is full message isn't exclusive to Rest failing, it's used for all recovery moves at full HP.

**Comment by Volco** [comment]
currently all recovery moves used at full hp just say "but it failed!" don't they?

**Comment by Zarel** [comment]
Oh, is it also displayed when Rest is used at full HP with no status?

**Comment by Volco** [comment]
i can check on my moon save file real fast

**Comment by Volco** [comment]
okay, i tested it on pokemon moon, recover does say it even when used at full HP with no status. meaning this becomes an entirely different issue

**Comment by DaWoblefet** [comment]
Not only should all recovery moves at full HP display "<Pokemon>'s HP is full!", including Heal Pulse, Pollen Puff, and Floral Healing, but Rest gets its message at a different point than all the other recovery moves. See SadisticMystic's [list of checks done before move execution](https://www.smogon.com/forums/threads/ultra-sun-ultra-moon-battle-mechanics-research-read-post-2.3620030/page-10#post-8096570); Rest at full HP is checked at step 28, while the other healing moves are checked at step 40.

**Comment by Volco** [comment]
The " [Pokemon]'s HP is Full!" message should definitely be added to to client regardless but not as a rest specific fail

**Comment by Zarel** [comment]
Agreed.

**Comment by Volco** [comment]
Tomorrow after classes i'll add the "fullHpFail" to the client and change rest to use it then make a new pr for the ones that also use it in-game, which i'll have to test before i make the changes to other moves

**Comment by Zarel** [comment]
`fullHpFail` is a bad name. Look at how the other messages are named.

**Comment by Zarel** [comment]
It should probably start with `fail`. `failFullHp` might work, but I think I'd rather put it in "misc effects" under `heal:fail`.

**Comment by Volco** [comment]
Sorry, i said i'd do the pr updates yesterday. I might not be until the weekend at this rate since i've had too get up earlier than normal this entire week so i've been too tired to get the updates done once i got home

**Comment by scheibo** [comment]
Don't worry about it - there isn't a timer on this! Take your time, thank you for your contributions. :)

**Comment by Zarel** [comment]
Should we put the other heal failure messages in this PR, too? What situations do they occur in?

**Comment by Volco** [comment]
I'm going to test that ingame when i have time so that may take a bit longer. I'll probably make a new pr for all of those

**Pull request closed without merging**